### ISSUE TITLE
WPEFramework rdk services error

### ISSUE THREAD

**Issue by sahil-siwatch** [Metrological/WebKitForWayland#694]
Hello. I was building WPE webkit image for rpi using buildroot. I get this error. I can't find my way out of this. How can this be 404 page ??
![image](https://user-images.githubusercontent.com/53403558/106733163-8bd29400-6637-11eb-8494-adaa5c0da0f9.png)

**Comment by nicoske** [comment]
Same issue here, the package now use git@github.com:WebPlatformForEmbedded/ThunderNanoServicesRDK.git
which requires an SSH key

**Comment by sahil-siwatch** [comment]
> Same issue here, the package now use [git@github.com](mailto:git@github.com):WebPlatformForEmbedded/ThunderNanoServicesRDK.git
> which requires an SSH key

yes. I guess this is the issue here. I also tried downloading the zip file of the now used rdk services. 
![image](https://user-images.githubusercontent.com/53403558/107602144-9e0f8c00-6c4e-11eb-9563-64223f225517.png)
https://github.com/rdkcentral/ThunderNanoServices and then added it to my personal git account the same manner as they are doing but again it showed the same issue. Next step: I added an ssh key in my local directory and added it to the git that i created. The error changed this time. I will try to get the screenshot soon but maybe you can try this as well and see what comes. Whatever the error was, it wasn't clear and i also searched for it. 
Is there any way to add the given package manually in the directories of buildroot?? might have to read the buildroot documentation.

**Comment by nicoske** [comment]
Buildroot tries to download the package as a zip file on their repo, obviously as it's an external git repository it's not available at buildroot side.
It seems that RDK closed down this repo and it's not publicly available anymore..

I modified the package to point to https://github.com/rdkcentral/rdkservices/ which is still available:
```
#WPEFRAMEWORK_RDKSERVICES_VERSION = a69018168069609bc91e2dab2b516fb6839d103c
WPEFRAMEWORK_RDKSERVICES_VERSION = 4492ff593ae9890f9d9ce42ef8737583268ef9c4
#WPEFRAMEWORK_RDKSERVICES_SITE_METHOD = git
#WPEFRAMEWORK_RDKSERVICES_SITE = git@github.com:WebPlatformForEmbedded/ThunderNanoServicesRDK.git
WPEFRAMEWORK_RDKSERVICES_SITE = $(call github,rdkcentral,rdkservices,$(WPEFRAMEWORK_RDKSERVICES_VERSION))
WPEFRAMEWORK_RDKSERVICES_INSTALL_STAGING = YES
WPEFRAMEWORK_RDKSERVICES_DEPENDENCIES = wpeframework-clientlibraries

```

**Comment by sahil-siwatch** [comment]
> Buildroot tries to download the package as a zip file on their repo, obviously as it's an external git repository it's not available at buildroot side.
> It seems that RDK closed down this repo and it's not publicly available anymore..
> 
> I modified the package to point to https://github.com/rdkcentral/rdkservices/ which is still available:
> 
> ```
> #WPEFRAMEWORK_RDKSERVICES_VERSION = a69018168069609bc91e2dab2b516fb6839d103c
> WPEFRAMEWORK_RDKSERVICES_VERSION = 4492ff593ae9890f9d9ce42ef8737583268ef9c4
> #WPEFRAMEWORK_RDKSERVICES_SITE_METHOD = git
> #WPEFRAMEWORK_RDKSERVICES_SITE = git@github.com:WebPlatformForEmbedded/ThunderNanoServicesRDK.git
> WPEFRAMEWORK_RDKSERVICES_SITE = $(call github,rdkcentral,rdkservices,$(WPEFRAMEWORK_RDKSERVICES_VERSION))
> WPEFRAMEWORK_RDKSERVICES_INSTALL_STAGING = YES
> WPEFRAMEWORK_RDKSERVICES_DEPENDENCIES = wpeframework-clientlibraries
> ```

I also tried this. Same response. Did it work for you ??

**Comment by woutermeek** [comment]
Could you please try to use our OSS Yocto recipes instead of buildroot?
https://github.com/WebPlatformForEmbedded/meta-wpe

**Comment by alvinshi** [comment]
Is there any update on this issue? I tried the OSS Yocto recipes too. But it did not work with the error message when trying to start the bitbake server: 

ERROR: Layer wpe-layer is not compatible with the core layer which only supports these series: hardknott honister (layer is compatible with dunfell warrior gatesgarth zeus thud sumo)
ERROR: Parse failure with the specified layer added, aborting.

**Comment by NPellet** [comment]
Isn't this package already covered by wpeservices-plugins, which references the new ThunderNanoServices repo in rdkcentral ?

**Comment by nicoske** [comment]
I'm lost with the new repo setup.. but I really would like to use buildroot as I did massive work using this framework for my project

**Comment by admindotnu** [comment]
Nothing changed. Current buildroot still gives the same error.

**Comment by jkey** [comment]
Same problem here as well. Would like to use buildroot for a current project. I could not achieve any success with Yocto so far. Would be nice if someone could help here.

**Comment by nicoske** [comment]
I'm still using buildroot but we're stuck using the last version of ThunderNanoServices published publicly on Github.
Still works with the latest build of WpeWebkit at https://github.com/WebPlatformForEmbedded/buildroot/commit/0d481aff67296bbfafd3cd323b77252ba04f8249

**Comment by jkey** [comment]
@nicoske thanks for the tip. I tried that but without success. Same message. I hope I do not bother you with it but do you have another idea? Thanks a lot for the help!

```
>> wpeframework-rdkservices 429dea8124310df9b7ca12436cd8a4e2521c607c Downloading
Reinitialized existing Git repository in /home/build/buildroot/dl/wpeframework-rdkservices/git/.git/
Fetching all references
ERROR: Repository not found.
fatal: Could not read from remote repository.

Please make sure that the correct access permissions are in place
and the repository exists.
Detected a corrupted git cache.
Removing it and starting afresh.
Empty git repository initialized in /home/build/buildroot/dl/wpeframework-rdkservices/git/.git/.
Fetching all references
ERROR: Repository not found.
fatal: Could not read from remote repository.

Please make sure that the correct access permissions are in place
and the repository exists.
Detected a corrupted git cache.
This is the second time in a row; bailing out
--2021-09-16 19:56:24-- http://sources.buildroot.net/wpeframework-rdkservices/wpeframework-rdkservices-429dea8124310df9b7ca12436cd8a4e2521c607c.tar.gz
Resolving the hostname sources.buildroot.net (sources.buildroot.net)... 104.26.1.37, 172.67.72.56, 104.26.0.37, ...
Connecting to sources.buildroot.net (sources.buildroot.net)|104.26.1.37|:80 ... connected.
HTTP request sent, waiting for response ... 404 Not Found
2021-09-16 19:56:24 ERROR 404: Not Found.

--2021-09-16 19:56:24-- http://sources.buildroot.net/wpeframework-rdkservices-429dea8124310df9b7ca12436cd8a4e2521c607c.tar.gz
Resolving the hostname sources.buildroot.net (sources.buildroot.net)... 172.67.72.56, 104.26.0.37, 104.26.1.37, ...
Connecting to sources.buildroot.net (sources.buildroot.net)|172.67.72.56|:80 ... connected.
HTTP request sent, waiting for response ... 404 Not Found
2021-09-16 19:56:25 ERROR 404: Not Found.

MAKE: *** [package/pkg-generic.mk:171: /home/build/buildroot/output/build/wpeframework-rdkservices-429dea8124310df9b7ca12436cd8a4e2521c607c/.stamp_downloaded] Error 1
```

**Comment by nicoske** [comment]
@jkey Here's the last commit that worked for me, from Nov 2020 in package/wpe/wpeframework-rdkservices:

```
WPEFRAMEWORK_RDKSERVICES_VERSION = 365ffa2779e5fef299e74e5a3db80f4dacbba896
WPEFRAMEWORK_RDKSERVICES_SITE = $(call github,rdkcentral,rdkservices,$(WPEFRAMEWORK_RDKSERVICES_VERSION))
WPEFRAMEWORK_RDKSERVICES_INSTALL_STAGING = YES
WPEFRAMEWORK_RDKSERVICES_DEPENDENCIES = wpeframework

```

**Comment by jkey** [comment]
@nicoske Thank you very much. Now I get the next error that the file ocdm/IOCDM.h is not found. Did I miss something?

```
In the file included from /home/build/buildroot/output/build/wpeframework-rdkservices-365ffa2779e5fef299e74e5a3db80f4dacbba896/OpenCDMi/CENCParser.h:23,
                 from /home/build/buildroot/output/build/wpeframework-rdkservices-365ffa2779e5fef299e74e5a3db80f4dacbba896/OpenCDMi/CENCParser.cpp:20:
/home/build/buildroot/output/build/wpeframework-rdkservices-365ffa2779e5fef299e74e5a3db80f4dacbba896/OpenCDMi/Module.h:31:10: fatal error: ocdm/IOCDM.h: file or directory not found
 #include <ocdm/IOCDM.h>
          ^~~~~~~~~~~~~~
Compilation finished.
make[3]: *** [OpenCDMi/CMakeFiles/WPEFrameworkOCDM.dir/build.make:89: OpenCDMi/CMakeFiles/WPEFrameworkOCDM.dir/CENCParser.cpp.o] Error 1
make[3]: *** Waiting for unfinished processes....
In the file included from /home/build/buildroot/output/build/wpeframework-rdkservices-365ffa2779e5fef299e74e5a3db80f4dacbba896/OpenCDMi/OCDM.h:23,
                 from /home/build/buildroot/output/build/wpeframework-rdkservices-365ffa2779e5fef299e74e5a3db80f4dacbba896/OpenCDMi/OCDM.cpp:20:
/home/build/buildroot/output/build/wpeframework-rdkservices-365ffa2779e5fef299e74e5a3db80f4dacbba896/OpenCDMi/Module.h:31:10: fatal error: ocdm/IOCDM.h: file or directory not found
 #include <ocdm/IOCDM.h>
          ^~~~~~~~~~~~~~
Compilation finished.
make[3]: *** [OpenCDMi/CMakeFiles/WPEFrameworkOCDM.dir/build.make:63: OpenCDMi/CMakeFiles/WPEFrameworkOCDM.dir/OCDM.cpp.o] Error 1
[65%] Creating the CXX object Monitor/CMakeFiles/WPEFrameworkMonitor.dir/MonitorJsonRpc.cpp.o
Contained in file from /home/build/buildroot/output/build/wpeframework-rdkservices-365ffa2779e5fef299e74e5a3db80f4dacbba896/OpenCDMi/OCDMJsonRpc.cpp:20:
/home/build/buildroot/output/build/wpeframework-rdkservices-365ffa2779e5fef299e74e5a3db80f4dacbba896/OpenCDMi/Module.h:31:10: fatal error: ocdm/IOCDM.h: file or directory not found
 #include <ocdm/IOCDM.h>

Translated with www.DeepL.com/Translator (free version)
```

**Comment by huesla** [comment]
I guess you run this from the same repo (https://github.com/WebPlatformForEmbedded/buildroot)
Trying to build an image for RPI3 I am hitting different errors, but I couldn't fix this one so far...
The repository seems to lack a bit of care and I am asking myself whether one is obliged to use Yocto for building a lightweight distro with WPE

**Comment by dschinnerl** [comment]
```
.../buildroot-wpe/output/build/wpeframework-rdkservices-R3.4$ git diff
diff --git a/package/wpe/wpeframework-rdkservices/wpeframework-rdkservices.mk b/package/wpe/wpeframework-rdkservices/wpeframework-rdkservices.mk
index af290daf6c..a0793d895a 100644
--- a/package/wpe/wpeframework-rdkservices/wpeframework-rdkservices.mk
+++ b/package/wpe/wpeframework-rdkservices/wpeframework-rdkservices.mk
@@ -3,9 +3,9 @@
 # wpeframework-rdkservices
 #
 ################################################################################
-WPEFRAMEWORK_RDKSERVICES_VERSION = a74f507795a6bf0e164de3c303d36b588064125b
+WPEFRAMEWORK_RDKSERVICES_VERSION = R3.4
 WPEFRAMEWORK_RDKSERVICES_SITE_METHOD = git
-WPEFRAMEWORK_RDKSERVICES_SITE = git@github.com:WebPlatformForEmbedded/ThunderNanoServicesRDK.git
+WPEFRAMEWORK_RDKSERVICES_SITE = https://github.com/rdkcentral/ThunderNanoServices
 WPEFRAMEWORK_RDKSERVICES_INSTALL_STAGING = YES
 WPEFRAMEWORK_RDKSERVICES_DEPENDENCIES = wpeframework-clientlibraries
```
Although I got another error afterwards ("No rule to make target 'install/fast'.  Stop.") I changed `Makefile` to call `preinstall` for `install/fast` too:

```
install/fast:
        $(MAKE) -f CMakeFiles/Makefile2 preinstall
.PHONY : install/fast
```

Maybe it helps someone (`raspberrypi3_wpe_defconfig`).

**Comment by dlbogdan** [comment]
trying to compile and get it working as it seems it's the only project that implements this browser fully accelerated using the binary firmware blobs instead of the opensource vc4 which is slower and takes more ram. 
I did compile it eventually, R3.5 to be exact, but I can't get it to boot on the rpi3. tried 3A+ and 3B 1.2. black screen. 
I'm trying to get R4.2 to work now. 
Also, I had to disable clearkeys or something. hope that isn't very important, I just want a fast accelerated browser in the end.
